rclcpp/topics/minimal_subscriber/CMakeLists.txt

```diff
@@ -36,7 +36,7 @@ ament_target_dependencies(subscriber_not_composable rclcpp std_msgs)
 add_executable(subscriber_content_filtering content_filtering.cpp)
 ament_target_dependencies(subscriber_content_filtering rclcpp std_msgs)
 
-add_library(wait_set_subscriber_library SHARED
```

Comment:
`rclcpp_components_register_node` below is registering this as a plugin. This can [only be used with shared libraries](https://github.com/ros2/rclcpp/blob/3d69031d2a614cf02dc10ea5db153471ea32b1f2/rclcpp_components/cmake/rclcpp_components_register_node.cmake#L21-L22). I don't think the problem in #344 is this library being a shared library; instead, I think there's a problem is in whatever is expecting it not to be one.